src/map/rodex.c

```diff
@@ -570,6 +588,14 @@ static void rodex_get_items(struct map_session_data *sd, int8 opentype, int64 ma
 		return;
 	}
 
+	// Queue the mail ID to be claimed
+	VECTOR_ENSURE(sd->rodex.claim_list, 1, 1);
+	VECTOR_PUSH(sd->rodex.claim_list, mail_id);
```

Comment:
When this function is called recursively by rodex_getItemsAck (while items are queued), this causes the next queued item to be inserted twice. I think the push should only happen when this is called directly, and not when this is (re)called during queue processing.

Also, if the second call to this function fails (because i.e. inventory has become full after retrieving one item and moving on to the second item in queue), queue processing gets aborted since rodex_getItemsAck is never called again, but we still have leftover items in the queue that will not be removed since the VECTOR_ERASE only happens there